web/app/js/components/Version.jsx

```diff
@@ -0,0 +1,75 @@
+import { ApiHelpers } from './util/ApiHelpers.js';
+import { Link } from 'react-router-dom';
+import React from 'react';
+import './../../css/version.css';
+
+export default class Version extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.loadFromServer = this.loadFromServer.bind(this);
+    this.handleApiError = this.handleApiError.bind(this);
+    this.state = {err: null, latest: null, loaded: false, pendingRequests: false,};
+  }
+
+  componentDidMount() {
+    this.loadFromServer();
+  }
+
+  loadFromServer() {
+    if (this.state.pendingRequests) {
+      return; // don't make more requests if the ones we sent haven't completed
+    }
+    this.setState({ pendingRequests: true });
+
+    let versionUrl = `https://versioncheck.conduit.io/version.json?version=${this.props.releaseVersion}?uuid=${this.props.uuid}`;
+    let versionFetch = ApiHelpers("").fetch(versionUrl);
+    // expose serverPromise for testing
+    this.serverPromise = Promise.all([versionFetch])
+      .then(([resp]) => {
+        this.setState({
+          latest: resp.version,
+          loaded: true,
+          pendingRequests: false,
+        });
+      }).catch(this.handleApiError);
+  }
+
+  handleApiError(e) {
+    this.setState({
+      loaded: true,
+      pendingRequests: false,
+      err: e
```

Comment:
nit: in other places we do this we use "error"  and I'd prefer us to be consistent